File: gems/decomplex/examples/zig/co-update.zig
const Node = struct {
    storage: i32,
    provenance: i32,
};

pub fn stable_one(node: *Node) void {
    node.storage = 1;
    node.provenance = 1;
}

pub fn stable_two(node: *Node) void {
    node.storage = 1;
    node.provenance = 1;
}

pub fn stable_three(node: *Node) void {
    node.storage = 1;
    node.provenance = 1;
}

pub fn misses_provenance(node: *Node) void {
    node.storage = 1;
}

File: gems/decomplex/examples/rust/co-update.rs
struct Node {
    storage: i32,
    provenance: i32,
}

fn stable_one(mut node: Node) {
    node.storage = 1;
    node.provenance = 1;
}

fn stable_two(mut node: Node) {
    node.storage = 1;
    node.provenance = 1;
}

fn stable_three(mut node: Node) {
    node.storage = 1;
    node.provenance = 1;
}

fn misses_provenance(mut node: Node) {
    node.storage = 1;
}
